merge_checks/merge_checks_runner.py

```diff
@@ -0,0 +1,69 @@
+import os
+import sys
+from dataclasses import dataclass
+from enum import Enum
+
+from merge_checks.commit_checks import get_commit_checks_result
+from merge_checks.commit_status_setter import set_commit_status
+
+
+@dataclass
+class Commit:
+    repository: str
+    commit_sha: str
+    token: str
+
+
+@dataclass
+class Status:
+    status_name: str
+    details_url: str
+
+
+class State(Enum):
+    pending = "pending"
+    success = "success"
+    failure = "failure"
+    error = "error"
+
+
+def get_commit_and_status() -> tuple[Commit, Status]:
+    commit = Commit(
+        repository=os.environ["GITHUB_REPOSITORY"],
+        commit_sha=os.environ["GITHUB_SHA"],
+        token=os.environ["TOKEN"],
+    )
+
+    status = Status(
+        status_name="Merge checks / Result",
+        details_url=f"{os.environ['GITHUB_SERVER_URL']}/{commit.repository}/actions/runs/{os.environ['GITHUB_RUN_ID']}",
+    )
+
+    return commit, status
+
+
+def run():
+
+    commit, status = get_commit_and_status()
+
+    set_commit_status(
+        **(commit.__dict__ | status.__dict__),
```

Comment:
```suggestion
        **(asdict(commit) | asdict(status)),
```

I wonder if it is a better idea to pass the `commit`  and `status` to the `set_commit_status` function.

Reply:
I have no opinion on passing stuff, but I would at least go for the `asdict` suggestion. I also reacted to the usage of `__dict__`, but then forgot about it :(